### ISSUE TITLE
storefrontUrl issue in API calls 

### ISSUE THREAD

**Issue by StanislovasRamanauskas** [shopware/frontends#54]
Faced an issue with register method in useUser composable. storefrontUrl param in API request is sliced and wrong.
![Screenshot from 2023-02-07 13-40-03](https://user-images.githubusercontent.com/38095299/217237029-27575be0-7ce0-404f-a679-6cf6db945fb7.png)
![Screenshot from 2023-02-07 13-41-52](https://user-images.githubusercontent.com/38095299/217237107-3c9126fb-cedc-4cf0-94d6-5c74188b6d22.png)

Found method which is responsible for storefrontUrl slicing - useInternationalization
![Screenshot from 2023-02-07 13-40-15](https://user-images.githubusercontent.com/38095299/217237542-cf52fd08-93fc-4f1e-b7ff-1fa74f1d7c73.png)

My nuxt config looks like this: 
![Screenshot from 2023-02-07 13-37-41](https://user-images.githubusercontent.com/38095299/217237617-d398ee6e-058f-457e-9c84-9d35d5cab2d6.png)

So few questions.
1. Why is the apiInstance.config.endpoint sliced? I guess I can solve this by adding / in the shopwareEndpoint end in nuxt config (http://localhost:88/) but I wonder why it was done that way? :thinking: 
2. 88 port is my Shopware endpoint and 3000 front-end. storefrontUrl is set to shopware endpoint url which is wrong (as you can see from error choices, BE expects storefrontUrl to be front url). Is this mean that BE/FE with different domains is not supported?

**Comment by patzick** [comment]
Thanks for reporting that!
We added it here: https://github.com/shopware/frontends/commit/12c6c300a4da19406968c753f680ffb5e0d49764#diff-5991ed4dc6e0bfd09caff830faee98c6d7fee213add8e7046b73579d0004a569R32 as a temporary workaround for multidomain issue.

@mdanilowicz could you check if that's been fixed on API side?

**Comment by patzick** [comment]
ok it should work now with the `canary` version. Specifically: https://www.npmjs.com/package/@shopware-pwa/composables-next/v/0.0.0-canary-20230209124459
@StanislovasRamanauskas can you confirm that so we'll release it?

connected commit: https://github.com/shopware/frontends/commit/50e74be52034d1947e273985f778e986f077db44

**Comment by StanislovasRamanauskas** [comment]
Thanks for the quick fix! 
But it seems to be fixing only part of the issue. 
Url is not trimmed now :tada: 
But shopwareEndpoint parameter (back-end url) from nuxt config is still passed to the storefrontUrl (which should be front end url). And error is returned.

Shouldn't getStorefrontUrl function
``` javascript
function getStorefrontUrl() 
    return apiInstance.config.endpoint ?? window.location.origin ?? "";
}
```

Be only like this (API call works correctly this way)? 
``` javascript
function getStorefrontUrl() 
    return window.location.origin ?? "";
}
```

Or maybe new parameter should be introduced for front-end url? I saw that in [#53](https://github.com/shopware/frontends/pull/53/files) @GM-Alex added browserEndpoint parameter which looks like meant for front-end url. But in that PR getStorefrontUrl function in not changed to use new param :thinking: 

Error which I receive on user registration again:
![Screenshot from 2023-02-09 16-55-55](https://user-images.githubusercontent.com/38095299/217852446-7cee80d6-193b-4c35-a71c-c5c561ffda74.png)

**Comment by mdanilowicz** [comment]
@StanislovasRamanauskas We cannot use `window.location.origin` because of cloud limitation and multi domains problems.

I have an idea of how to resolve your problem, but I need to understand one thing:

localhost:3000 is your frontends app application URL, and your storefront configuration is also localhost:3000?
![Screenshot 2023-02-09 at 17 46 13](https://user-images.githubusercontent.com/7597086/217881618-b3ba2cbb-d8d8-4938-911c-99d6699f3213.png)

What is the reason for having it on the same port? Can you have the storefront url on :88 port?

**Comment by GM-Alex** [comment]
@StanislovasRamanauskas Within my PR there is an issue because I thought the browser URL feature is part of axios but is part of the axios module for nuxt. But if @patzick would agree to implement a separate and optional browser URL, I would adjust it. The idea is to do it like the module implementation, see https://axios.nuxtjs.org/options#baseurl

This would give us full flexibility and allows us to route SSR traffic through an internal network, for example to skip a WAF or for other tweeks. In addition it also makes docker setups a little more comfortable.

**Comment by StanislovasRamanauskas** [comment]
@mdanilowicz
yes, localhost:3000 is my frontends application URL. And in Shopware I have one sales channel which domain is pointed to frontends app (localhost:3000).
![Screenshot from 2023-02-10 00-06-10](https://user-images.githubusercontent.com/38095299/217950031-ef59495f-b6c6-40e7-a280-d2b47497f2df.png)
So you're saying simply change it to localhost:88? :thinking: Wouldn't it create some kind of URL incompatibility issues with Frontends? Tried it and one thing noticed from the start is that now "Open storefront" button opens native Shopware twig storefront :thinking:  
![Screenshot from 2023-02-10 00-12-24](https://user-images.githubusercontent.com/38095299/217952070-ee59d1c5-121f-45c0-a126-a505ca050c8b.png)

**Comment by sanis** [comment]
It doesn't really make sense to setup API link as a domain in the backend, because it should be channel connected (one backend can be consumed by many frontend links). It could work as a workaround, but in the long run, according to storefrontUrl should be passed to the API.

**Comment by mdanilowicz** [comment]
@StanislovasRamanauskas Please remember that if you use the frontends app, you are not creating a storefront. Frontends are a "headless" app.

So please create a headless sales channel and use a `localhost:88` address. We are working on the documentation to describe how it should be set up from the admin panel. 

Also, `storefrontUrl` does not make sense when talking about headless apps, that is why we are thinking about some changes in that area with the internalisation feature

**Comment by StanislovasRamanauskas** [comment]
@mdanilowicz thanks for answers, you are right, we need to use headless. Tried and it solves the issue and everything seems to works fine.
I guess I'll wait for more info in docs and will watch storefrontUrl future changes.
